### ISSUE TITLE
supports Django 1.11

### ISSUE THREAD

**Issue by dubirajara** [radiac/django-tagulous#28]
hi, does not support the new django 1.11 version? 
i have this problem:

> ERRORS:
core._Tagulous_Ideas_tags: (models.E023) The model name '_Tagulous_Ideas_tags' cannot start or end with an underscore as it collides with the query lookup syntax.
The command "python manage.py migrate" failed and exited with 1 during .

thanks

**Comment by radiac** [comment]
I don't think anybody has started work on 1.11 support yet - I'll update this issue when there is news.

**Comment by theskumar** [comment]
Django 1.11 is here 🎉   Wondering if we get this in! :)

**Comment by tomasgarzon** [comment]
Hi, when do you think we will have the next release with this issue resolved?

**Comment by dubirajara** [comment]
hi @radiac , i fork this project and i change the Prefix for models auto-generated by tagulous, i remove MODEL_PREFIX  "undercore _" to work django versión 1.11, now its work fine with Django 1.11
Can I have a problem with that change I made?

**Comment by ricosalomar** [comment]
An easy fix is to use a custom tag model, a la:
```
class TagModel(tagulous.models.TagModel):
    class TagMeta:
        pass
```
```
class MyModel(models.Model):
    ....
    tags = tagulous.models.TagField(to=TagModel)
```

That way tagulous doesn't need to auto-generate model names.

**Comment by dubirajara** [comment]
@ricosalomar its not work to me.

**Comment by ricosalomar** [comment]
Well, looks like you should check to see which tag options you are setting.
Also, give it a better name, like `MyTagModel`.
I believe you need to set them in `MyTagModel.TagMeta` and not in the field.
E.g.:
```
class MyTagModel(tagulous.models.TagModel):
    class TagMeta:
        autocomplete_view = 'tags_autocomplete'
```
And the field declaration needs to have no options, except `to=...`
```
tags = tagulous.models.TagField(to=MyTagModel)
```

**Comment by radiac** [comment]
Sorry for the delay getting on to this - @dubirajara is right, the main issue seems to be the leading `_` in the model prefix, and removing that seems to get everything working. As @ricosalomar says, a custom tag model will get around that problem altogether (and yes, tag options on custom tag models have to be set on the model itself - otherwise you could end up with two tag fields pointing at the same custom tag model with different options, which would lead to madness).

I've now got a working branch with experimental Django 1.11 support - that is to say some tests are failing, but they don't seem to be failing in a way which should have much of an impact. At least I think - hence "experimental".

I'm really busy irl at the moment, but hope to get the remaining failing tests fixed and Tagulous 0.13.0 out shortly.

In the meantime, you should be able to install the branch with:

    pip install -e git+https://github.com/radiac/django-tagulous.git@feature/django-1_11#egg=django-tagulous

Be sure to read the [upgrade instructions](https://github.com/radiac/django-tagulous/blob/feature/django-1_11/docs/upgrading.rst#upgrading-from-0120) for this next version though - because the prefix has changed, you'll need to create migrations to rename your auto-generated tag models.

**Comment by metaljoe** [comment]
Hi. I'm upgrading a Django application to 1.11 and had the same problem. Is there an update on when 0.13.0 will be available?

Oh, and thanks for this library - it's been really useful!

**Comment by radiac** [comment]
0.13.0 has now been released, supporting Django 1.11.

Thanks for your patience everyone - life has been hectic!

**Comment by metaljoe** [comment]
Big thank you for releasing this 👍
